### PR TITLE
Remove allowed keys list for adding woocommerce_meta data

### LINKED ISSUE #6888: Extending the user's WooCommerce meta fields
### Description

While working on a new analytics page we encountered an issue regarding the user's WooCommerce meta. Turns out that we cannot currently add a custom meta key and every action ends up with an error message:
```
No valid woocommerce_meta keys were provided for update.
```

Currently, we use the `useUserPreferences()` module  for updating user preferences regarding the columns visiblity in custom report pages. This would require to use the module like so:
```javascript
import { useUserPreferences } from '@woocommerce/data';

// [...]

const { updateUserPreferences, ...userData } = useUserPreferences();

// [...] 

const userDataFields = {
    [ 'custom_report_page_meta_key' ]: hiddenColumns,
};

// Try to save.
updateUserPreferences( userDataFields )
```

After some digging, we found that this isn't consistent with the backend side of things. 
In PHP we do have available the `woocommerce_admin_get_user_data_fields` filter, that will add and load every custom key we need. Please see `Loader::get_user_data_fields` @ https://github.com/woocommerce/woocommerce-admin/blob/main/src/Loader.php#L1273 for reference.

### Solution
As this module is publicly available for 3rd parties through the `@woocommerce/data` package, we most probably need to introduce a new filter in the JS part that will allow custom meta keys. A new filter _(e.g. "woocommerce_user_data_fields")_ should be added in https://github.com/woocommerce/woocommerce-admin/blob/main/packages/data/src/user/use-user-preferences.js#L52

### PR DESCRIPTION
**Pull request by xristos3490**
Fixes #6888 

For description please check the issue details.

### Changelog entry
```
Dev: Removed allowed keys list for adding woocommerce_meta data. #6889 
```

_Note: Package changelog updated as well._

### CODE REVIEW

**Review by jeffstieler — CHANGES_REQUESTED** [review]
Hi @xristos3490,

Thank you for highlighting this issue and submitting a PR!

Rather than adding a filter, I think we should just remove the allowlist functionality. It seems to have little value if the list is filterable anyhow.

If you're not comfortable making those changes let me know and I'll take care of it.

Thanks again.

**Comment by xristos3490** [comment]
Hey @jeffstieler,

> It seems to have little value if the list is filterable anyhow.

Agreed! Taking a look at the additional user schema for "woocommerce_meta" added by the `Loader::register_user_data()` it seems that the validity of the individual meta keys are also checked in PHP. 
See:
https://github.com/woocommerce/woocommerce-admin/blob/a437d32d22d582fdbc03250e3a117caeb760b891/src/Loader.php#L1251

With that said, there is no way to "spam" with meta keys without first declaring them through the `woocommerce_admin_get_user_data_fields` filter.

Moreover, updating a filter for the "_Allowlist_" in both places should be hard to maintain and therefore removing this filter should keep our data in a **single source of truth**.

File: packages/data/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Unreleased
 
 -   Fix parsing bad JSON string data in useUserPreferences hook. #6819
+-   Removed allowed keys list for adding woocommerce_meta data. #6889
```
Comment (jeffstieler) [inline packages/data/CHANGELOG.md:4]: Can you add this line to `readme.txt` under the `== Unreleased ==` header? That will stop the linter from failing.

Like:
`- Dev: Removed allowed keys list for adding woocommerce_meta data. #6889`

**Review by jeffstieler — CHANGES_REQUESTED** [review]
This is testing well, but the linters aren't happy. 😅 

Can you add the change log entry (I pointed out where inline), and fix the code formatting issues?

* Remove the unused `pick` import
* Change the `mapValues()` call to be a single line

You can run the linters locally using `npm run lint:js`.

**Review by jeffstieler — APPROVED** [review]
Nice work, thank you for the contribution!

**Pull request merged**